<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>593</width>
    <height>601</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="MyGLWidget" name="myGLWidget" native="true">
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QFrame" name="frame">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>150</height>
       </size>
      </property>
      <property name="baseSize">
       <size>
        <width>0</width>
        <height>0</height>
       </size>
      </property>
      <property name="frameShape">
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow">
       <enum>QFrame::Raised</enum>
      </property>
      <layout class="QGridLayout" name="grid">
       <item row="0" column="2">
        <widget class="QSlider" name="rotXSlider">
         <property name="maximum">
          <number>359</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="lblYrotation">
         <property name="text">
          <string>Y Rotation</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QLabel" name="lblZrotation">
         <property name="text">
          <string>Z Rotation</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QSlider" name="rotZSlider">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>359</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="lblXrotation">
         <property name="text">
          <string>X Rotation</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QSlider" name="rotYSlider">
         <property name="minimum">
          <number>0</number>
         </property>
         <property name="maximum">
          <number>359</number>
         </property>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="whiteLightButton">
         <property name="text">
          <string>White Light</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
         <property name="autoDefault">
          <bool>false</bool>
         </property>
         <property name="default">
          <bool>false</bool>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QComboBox" name="accuracyCombo">
         <property name="minimumSize">
          <size>
           <width>130</width>
           <height>0</height>
          </size>
         </property>
         <property name="currentText">
          <string>20 triangles</string>
         </property>
         <property name="insertPolicy">
          <enum>QComboBox::NoInsert</enum>
         </property>
         <item>
          <property name="text">
           <string>20 triangles</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>80 triangles</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>320 triangles</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1280 triangles</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="yellowLightButton">
         <property name="text">
          <string>Yellow Light</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="checked">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QComboBox" name="shadingCombo">
         <item>
          <property name="text">
           <string>Flat Shading</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Smooth Shading</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">myglwidget.h</header>
   <container>1</container>
   <slots>
    <slot>setXRotation(int)</slot>
    <slot>setYRotation(int)</slot>
    <slot>setZRotation(int)</slot>
    <slot>setAccuracy(int)</slot>
    <slot>setWhiteLight(bool)</slot>
    <slot>setYellowLight(bool)</slot>
    <slot>setSmoothShading(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>rotYSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>myGLWidget</receiver>
   <slot>setYRotation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>346</x>
     <y>506</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>213</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotZSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>myGLWidget</receiver>
   <slot>setZRotation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>333</x>
     <y>527</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>187</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotXSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>myGLWidget</receiver>
   <slot>setXRotation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>472</y>
    </hint>
    <hint type="destinationlabel">
     <x>155</x>
     <y>330</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>whiteLightButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>myGLWidget</receiver>
   <slot>setWhiteLight(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>78</x>
     <y>469</y>
    </hint>
    <hint type="destinationlabel">
     <x>88</x>
     <y>388</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yellowLightButton</sender>
   <signal>toggled(bool)</signal>
   <receiver>myGLWidget</receiver>
   <slot>setYellowLight(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>511</y>
    </hint>
    <hint type="destinationlabel">
     <x>68</x>
     <y>358</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>accuracyCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>myGLWidget</receiver>
   <slot>setAccuracy(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>52</x>
     <y>533</y>
    </hint>
    <hint type="destinationlabel">
     <x>47</x>
     <y>366</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shadingCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>myGLWidget</receiver>
   <slot>setShading(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>117</x>
     <y>562</y>
    </hint>
    <hint type="destinationlabel">
     <x>124</x>
     <y>377</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>shadingCombo</sender>
   <signal>currentIndexChanged(int)</signal>
   <receiver>myGLWidget</receiver>
   <slot>setSmoothShading(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>91</x>
     <y>564</y>
    </hint>
    <hint type="destinationlabel">
     <x>111</x>
     <y>358</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
